arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/cancelBoundary.kt

```diff
@@ -6,6 +6,28 @@ import kotlin.coroutines.intrinsics.suspendCoroutineUninterceptedOrReturn
 /**
  * Checks for cancellation,
  * when cancellation occurs this coroutine will suspend indefinitely and never continue.
```

Comment:
The example is clear ... but, the textual explanation does not seem sufficient to understand the purpose of cancelBoundary

Reply:
What about the following instead?  I found this terribly hard to document.

"Inserts a cancellable boundary — it checks for the cancellation status of the suspend-loop and does not allow the continuation to keep executing in the case cancellation happened."

Reply:
Maybe:

In a cancelable environment, we need to add mechanisms to react when a cancellation is triggered : a cancel boundary inserts checking point in suspendable methods. At runtime, a cancel boundary checks for the cancellation status of the suspend-loop and does not allow the continuation to keep executing in the case a cancellation was triggered.

Or, something like that

Reply:
Thank you @PhBastiani! That is a much better improvement over what I did 😅